scripts/cleanup_ebpf_cicd_tests.ps1

```diff
@@ -15,11 +15,19 @@ $TestVMCredential = Get-StoredCredential -Target $Target -ErrorAction Stop
 # Load other utility modules.
 Import-Module .\common.psm1 -Force -ArgumentList ($LogFileName) -WarningAction SilentlyContinue
 Import-Module .\config_test_vm.psm1 -Force -ArgumentList ($TestVMCredential.UserName, $TestVMCredential.Password, $WorkingDirectory, $LogFileName) -WarningAction SilentlyContinue
+Import-Module .\install_ebpf.psm1 -ArgumentList ($WorkingDirectory, $LogFileName) -Force -WarningAction SilentlyContinue
 
 # Read the test execution json.
 $TestExecutionConfig = Get-Content ("{0}\{1}" -f $PSScriptRoot, $TestExecutionJsonFileName) | ConvertFrom-Json
 $VMList = $TestExecutionConfig.VMMap.$SelfHostedRunnerName
 
+# Uninstall eBPF Components on the test VM.
+foreach($VM in $VMList) {
+       $VMName = $VM.Name
+       Write-Host "Uninstalling eBPF components on VM $VMName..."
+       Uninstall-eBPFComponentsOnVM -VMName $VMname -ErrorAction Stop
```

Comment:
I wonder if we could stop restoring the checkpoint in this step, now that we're uninstalling eBPF.
This was done to avoid having lingering ebpf bits when someone later works on a runner and accidentally pollutes the checkpoint. 
Not restoring Restore-AllVMs here would allow us to eg. manually grab a dump or traces that failed to upload, as long as another job hasn't been scheduled on a given VM yet.

Reply:
I'd not make this change as part of this PR, which is solely aimed at using the MSI in the setup process.

It could though be a good proposal you could file in a dedicated issue, for a more generic scope in designing the GH runner handling, such also your above comment in squashing commits in PRs.

Reply:
Adding to this, we *must* restore the baseline, as the installation process (as also enforced by later comments) will throw exceptions in case of failures, to quickly stop for troubleshooting without running the rest unnecessarily.
Therefore, restoring the baseline for cleaning up all is paramount.

Reply:
This can certainly be done in a follow-up PR but I'm also strongly in favor of avoiding restoring the baseline as part of the post-test clean-up.  I agree with @rectified95 that this allows us to get to the dumps manually in case we fail to capture them for whatever reason.  We should leave the vm state 'as-is' after the post-test clean-up and should restore the baseline _before_ we initiate _anything_ on the VM the next time around.